plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraryContainerInitializer.java

```diff
@@ -121,7 +121,9 @@ private void validateJarPaths(LibraryClasspathContainer container) throws Librar
     IClasspathEntry[] classpathEntries = container.getClasspathEntries();
     for (int i = 0; i < classpathEntries.length; i++) {
       IClasspathEntry classpathEntry = classpathEntries[i];
-      if (!classpathEntry.getPath().toFile().exists()) {
+      if (!classpathEntry.getPath().toFile().exists()
+          || (classpathEntry.getSourceAttachmentPath() != null
+              && !classpathEntry.getSourceAttachmentPath().toFile().exists())) {
         classpathEntries[i] = repositoryService.rebuildClasspathEntry(classpathEntry);
```

Comment:
Future Speculation: I wonder if this wouldn't be better put into the repositoryService somehow — just thinking that the repository service would want to control how many of these operations are in progress at once.

Then again, it could use a `JobGroup` to manage the family of jobs.  That lets the Jobs framework manage parallelism.